modules/library/main/src/main/java/org/geotools/styling/Rule.java

```diff
@@ -172,4 +174,19 @@
 
     /** Used to traverse the style data structure. */
     void accept(org.geotools.styling.StyleVisitor visitor);
+
+    /** Determines if a vendor option with the specific key has been set on this Rule. */
+    default boolean hasOption(String key) {
```

Comment:
Work on the rule API change should go in a different ticket, as it's backed by a GSIP:
https://osgeo-org.atlassian.net/browse/GEOT-6870
(I did not create a dedicated one along with the proposal, my fault)

Reply:
fixed pr available here https://github.com/geotools/geotools/pull/3457

Reply:
Ok cool. Make sure to merge that one before and amend this one accordingly.

Reply:
fixed